gapic/src/main/com/google/gapid/views/PipelineView.java

```diff
@@ -184,202 +192,293 @@ protected void onUiThreadError(Loadable.Message error) {
+
+        for (int stageIndex = 0; stageIndex < stages.size(); stageIndex++) {
+          API.Stage stage = stages.get(stageIndex);
+          stageMap.put(stage.getDebugName(), createStage(stageComposite, stage));
+
+          Button stageButton = new Button(stripComposite, SWT.FLAT);
+          stageButton.setText(stage.getDebugName());
+          stageButton.addListener(SWT.Selection, e -> {
+            stageStack.topControl = stageMap.get(stage.getDebugName());
+            stageComposite.layout();
+            selectedStage = stage.getDebugName();
+            stripComposite.redraw();
+          });
+
+          stageButton.setBackground(getDisplay().getSystemColor(SWT.COLOR_LIST_BACKGROUND));
+          stageButton.setFont(theme.bigBoldFont());
+          stageButton.setToolTipText(stage.getStageName() + " (" + stage.getDebugName() + ")");
 
-        for (API.Stage stage : stages) {
-          TabItem item = createStandardTabItem(folder, stage.getDebugName());
+          if (!stage.getEnabled()) {
+            stageButton.setEnabled(false);
+            stageButton.addListener(SWT.Paint, e -> {
+              Rectangle areaSize = stageButton.getBounds();
+              e.gc.setBackground(getDisplay().getSystemColor(SWT.COLOR_LIST_FOREGROUND));
+              e.gc.setLineWidth((int)(areaSize.height * 0.1));
+              e.gc.drawLine(0, areaSize.height / 2, areaSize.width, areaSize.height / 2);
+              e.gc.drawPolygon(new int[] {0, 0, areaSize.width - 1, 0, areaSize.width - 1, areaSize.height - 1, 0, areaSize.height - 1});
+            });
+          } else {
+            stageButton.addListener(SWT.Paint, e -> {
+              Rectangle areaSize = stageButton.getBounds();
+              e.gc.setLineWidth((int)(areaSize.height * 0.1));
+
+              if (hoveredButton == stageButton) {
+                RGB hoverColor = getDisplay().getSystemColor(SWT.COLOR_BLUE).getRGB();
+                stageButton.setBackground(new Color(stageButton.getDisplay(), hoverColor.red+30, hoverColor.green+30, hoverColor.blue, 63));
+              } else {
+                stageButton.setBackground(getDisplay().getSystemColor(SWT.COLOR_LIST_BACKGROUND));
+              }
 
-          Composite stageGroup = createComposite(folder, new GridLayout());
+              if (hoveredButton == stageButton || (selectedStage != null && selectedStage.equals(stageButton.getText()))) {
+                e.gc.setForeground(getDisplay().getSystemColor(SWT.COLOR_BLUE));
+              } else {
+                e.gc.setBackground(getDisplay().getSystemColor(SWT.COLOR_LIST_FOREGROUND));
+              }
+              e.gc.drawPolygon(new int[] {0, 0, areaSize.width - 1, 0, areaSize.width - 1, areaSize.height - 1, 0, areaSize.height - 1});
+            });
+
+            stageButton.addListener(SWT.MouseEnter, e -> {
+              hoveredButton = stageButton;
+              stripComposite.redraw();
```

Comment:
This is the right place to change the button background color. No need to redraw the strip, which doesn't work anyway, because it does not redraw all children.
So replace the line with:
```stageButton.setBackground(hoverColor);```
You have to create the hoverColor when initializing the pipeline view, and dispose it properly.
Also please interpolate the hoverColor to be halfway between blue and SWT.COLOR_LIST_BACKGROUND.